src/csync/csync.cpp

```diff
@@ -51,42 +51,11 @@
 #include "csync_rename.h"
 #include "c_jhash.h"
 
-static int _key_cmp(const void *key, const void *data) {
-  uint64_t a;
-  csync_file_stat_t *b;
-
-  a = *(uint64_t *) (key);
-  b = (csync_file_stat_t *) data;
-
-  if (a < b->phash) {
-    return -1;
-  } else if (a > b->phash) {
-    return 1;
-  }
-
-  return 0;
-}
-
-static int _data_cmp(const void *key, const void *data) {
-  csync_file_stat_t *a, *b;
-
-  a = (csync_file_stat_t *) key;
-  b = (csync_file_stat_t *) data;
-
-  if (a->phash < b->phash) {
-    return -1;
-  } else if (a->phash > b->phash) {
-    return 1;
-  }
-
-  return 0;
-}
-
 void csync_create(CSYNC **csync, const char *local) {
   CSYNC *ctx;
   size_t len = 0;
 
-  ctx = (CSYNC*)c_malloc(sizeof(CSYNC));
+  ctx = new CSYNC{};
```

Comment:
c_malloc was zeroing some fields. 

 I guess we should put some of the code bellow in the constuctor or in the the struct itself, in the form of  `=0` and so on.

Reply:
Where is that zeroing happening?

Reply:
Ahh, `calloc` does. The `CSYNC{}` syntax did zero fields since it doesn't have an explicit default constructor, but I'll add normal constructors/destructors.

Reply:
Added a constructor/destructor in 5fb559c